src/sql/src/plan/statement/ddl.rs

```diff
@@ -602,6 +604,76 @@ pub fn plan_create_source(
                     UnplannedSourceEnvelope::Upsert(UpsertStyle::Debezium { after_idx })
                 }
                 DbzMode::Plain => {
+                    // XXX(chae): should this be for upsert too?
+                    // TODO(chae): validate shape of tx_metadata item: maybe including full dbz dedup?
+                    let tx_metadata = match with_options.remove("tx_metadata") {
+                        Some(Value::String(src)) => {
+                            // XXX(chae): this seems like I'm breaking a boundary by accessing scx.catalog
```

Comment:
@jkosh44 do you mind taking a quick look at this?  Not sure if I'm doing this the right way -- or if there's a better way to convert `src` into `item`

Reply:
In terms of looking up an item in the catalog, getting the full_name, and getting the description it all looks correct to me. However it make me a little nervous to see you hard-code `"materialize.public"` as the database and schema name. I don't really know the context of this PR, but it is possible to delete the `materialize` database and the `public` schema and things are supposed to continue to work as normal.

Reply:
If you don't know what database and schema `src` lives in, then just use `UnresolvedObjectName::unqualified` and the catalog has a search process to look for the item in various databases and schemas. Include `materialize.public` if it exists.

Reply:
Yeah I didn't like `materialize.public` either -- the issue I had with doing `UnresolvedObjectNmae::unqualified(&src)` is that when restarting, I get an error that `$whatever_was_in_src` is an unknown catalog item -- with the error stemming from here

Do you think this might be a good case for purification to do the resolution?

Reply:
tbh honest I don't exactly know what purification is. Is `src` already a qualified string? If so you might have to split it on periods. If you give me a way to re-create that error I can take a look at it. It's odd that it wouldn't be able to find it because `resolve_item` will look in `materialize.public` if it exists.

Reply:
We chatted in slack and I've changed the approach slightly to make sure we can handle the initial `CREATE SOURCE` call as well as its recreation after restarts:
- let `resolve_item` do it's thing on the initial call
- use normalize::create_statement to persist this normalization (often prepending `materialize.public.`) so that it's used on subsequent restarts

This normalization mirrors how we normalize the name of the source itself